### PR TITLE
Change name of cromwell helm chart to allow the name to be used as part of the templatized names (BW-859, BW-865)

### PR DESCRIPTION
**Pull request by cahrens**
This renames the Cromwell helm chart. The related cromwhelm MR is https://github.com/broadinstitute/cromwhelm/pull/9.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/main/resources/reference.conf [hunk 1]
```diff
@@ -413,8 +413,8 @@ gke {
   }
   cromwellApp {
     # If you update the chart name or version here, make sure to also update it in the dockerfile:
-    chartName = "/leonardo/CromwellHsqldbLocalMinikube"
-    chartVersion = "0.1.2"
+    chartName = "/leonardo/cromwell"
+    chartVersion = "0.1.3"
     services = [
       {
         name = "cromwell-service"
```
Comment (cahrens) [inline http/src/main/resources/reference.conf:420]: I do not believe that these services values are referenced anywhere (the Cromwell service is no longer named "cromwell-service", and in fact the name cannot be stated here because it starts with a randomly generated prefix). I would recommend that we remove this services block to reduce confusion.

File: Dockerfile [hunk 2]
```diff
@@ -46,7 +46,7 @@ RUN helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx && \
     helm repo add galaxy https://raw.githubusercontent.com/cloudve/helm-charts/anvil/ && \
     helm repo add terra-app-setup-charts https://storage.googleapis.com/terra-app-setup-chart && \
     helm repo add terra https://terra-app-charts.storage.googleapis.com && \
-    helm repo add cromwell-local https://broadinstitute.github.io/cromwhelm/charts/ && \
+    helm repo add cromwell-helm https://broadinstitute.github.io/cromwhelm/charts/ && \
```
Comment (cjllanwarne) [inline Dockerfile:49]: I see you stopped short of `cromwhelm` here... 😢

**Review by cjllanwarne — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (cahrens) [reply Dockerfile:49]: You know, this morning I was thinking that I should have just used cromwhelm. :)

File: http/src/main/resources/reference.conf (hunk 1 above)
Comment (cahrens) [reply http/src/main/resources/reference.conf:420]: @cjllanwarne what did you think of this ^? I'm wondering if you could remove them in your MR.

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
Since this MR requires the change to the cromwhelm repo in order for the leo build to pass (example failure: https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-build/12004/console), I will have to rerun automation tests after broadinstitute/cromwhelm#9 merges.

File: http/src/main/resources/reference.conf (hunk 1 above)
Comment (cjllanwarne) [reply http/src/main/resources/reference.conf:420]: Did you test bringing the app up? It looks like we're relying on these values for the "did Cromwell come up yet" check in Leo

Comment (cahrens) [reply http/src/main/resources/reference.conf:420]: I did test bringing the app up and was able to load the Cromwell Swagger page.

Comment (cjllanwarne) [reply http/src/main/resources/reference.conf:420]: It looks like we also use this value for generating the proxy path. It's possible that this is a "Leonardo service" name and has to _exist_ but doesn't need to match any underlying k8s service. I wonder whether @rtitle knows?

Comment (cahrens) [reply http/src/main/resources/reference.conf:420]: I was wondering if it is supposed to match a label? Here are the labels on the cromwell service.

image

Comment (cahrens) [reply http/src/main/resources/reference.conf:420]: I noticed that the Galaxy app has "galaxy" specified for services.name, but there isn't a service named galaxy.

Comment (cahrens) [reply http/src/main/resources/reference.conf:420]: Ah, I understand now what you are saying @cjllanwarne. It's here in the URL, but doesn't seem to need to match any of the k8s service names. https://leonardo-fiab.dsde-dev.broadinstitute.org:30443/proxy/google/v1/apps/terra-test-d180e87d/cromwell-1/cromwell-service/swagger/index.html?url=/proxy/google/v1/apps/terra-test-d180e87d/cromwell-1/cromwell-service/swagger/cromwell.yaml#/Engine/engineVersion

File: Dockerfile [hunk 3]
```diff
@@ -59,7 +59,7 @@ RUN cd /leonardo && \
     helm pull galaxy/galaxykubeman --version $GALAXY_VERSION --untar && \
     helm pull terra/terra-app --version $TERRA_APP_VERSION --untar  && \
     helm pull ingress-nginx/ingress-nginx --version $NGINX_VERSION --untar && \
-    helm pull cromwell-local/CromwellHsqldbLocalMinikube --version $CROMWELL_CHART_VERSION --untar && \
+    helm pull cromwell-helm/cromwell --version $CROMWELL_CHART_VERSION --untar && \
```
Comment (rtitle) [inline Dockerfile:62]: I see this in the [build](https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-build/12004/console):
```
11:35:18 Error: chart "cromwell" matching 0.1.3 not found in cromwell-helm index. (try 'helm repo update'): no chart name found
```

Does the chart exist yet?

Comment (cahrens) [reply Dockerfile:62]: The cromwhelm merge request has been approved but not yet merged. Once this ticket has thumbs I was going to merge cromwhelm, do jenkins retest here, and then merge this one after tests pass.

Comment (cahrens) [reply Dockerfile:62]: FWIW, I did deploy this branch to a fiab and the build passed when cromwhelm's pages value was set to my branch instead of main.

File: http/src/main/resources/reference.conf (hunk 1 above)
Comment (rtitle) [reply http/src/main/resources/reference.conf:420]: The idea is that the service name in the Leo config is used for setting the ingress rule and building the proxy URL to the app. It can actually be any string. The config is used as follows:

1. In front Leo, when we receive a `createApp` request, we read the serviceName from config and persist it to the DB. Looks like the cromwell case is currently handled, so we are persisting `cromwell-service` to the DB: https://github.com/DataBiosphere/leonardo/blob/f9007d5b47e9a48be254aced5ae9dc45315b6732/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala#L596-L603

2. In Back Leo, when we process a `createApp` request, we read the serviceName from the DB, and pass it to the ingress config via helm. Looks like we are currently hard-coding `"cromwell-service"`: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala#L1234. I think I would recommend reading the service from the DB instead (e.g. adding a `ServiceName` parameter to this method and passing it from the DB object).

3. After the app is created, when a `getApp` request is issued, Leo builds the proxyUrl to return to the caller. Looks like today we are reading the `serviceName` from the DB which is good, and appending some extra stuff to deal with Cromwell swagger.

So TLDR I think that config is used and we should keep it -- however it might be good to update `GKEInterepreter` to not hard-code `cromwell-service` and read it from the DB instead.

File: Dockerfile (hunk 3 above)
Comment (rtitle) [reply Dockerfile:62]: 👍

File: http/src/main/resources/reference.conf (hunk 1 above)
Comment (cahrens) [reply http/src/main/resources/reference.conf:420]: Thanks @rtitle. But to be clear, there is no need for "cromwell-service" to be the name that is in the helm chart for the Cromwell k8s service, correct? That used to be the name in our helm chart, but I am changing it to be based off of the app name, similar to galaxy.

**Review by rtitle — APPROVED** [review]
Looks good once helm chart is pushed and tests pass

**Comment by cahrens** [comment]
jenkins retest

File: http/src/main/resources/reference.conf (hunk 1 above)
Comment (rtitle) [reply http/src/main/resources/reference.conf:420]: Ah - yeah I think that's correct, this does _not_ need to be the name of the k8s service, it's just used for generating the ingress path.
